src/Compilers/CSharp/Test/Emit/Emit/DesktopStrongNameProviderTests.cs

```diff
@@ -31,6 +31,46 @@ public void RespectNullTempPath()
             Assert.Null(provider.FileSystem.GetSigningTempPath());
         }
 
+        [Fact]
+        public void EqualityUsingPath()
+        {
+            var tempDir = Temp.CreateDirectory();
+            var provider1 = new DesktopStrongNameProvider(tempPath: tempDir.Path);
+            var provider2 = new DesktopStrongNameProvider(tempPath: tempDir.Path);
+
+            Assert.Equal(provider1, provider2);
+        }
+
+        [Fact]
+        public void EqualityUsingKeyFileSearchPaths()
+        {
+            var tempDir = Temp.CreateDirectory();
+            var provider1 = new DesktopStrongNameProvider(keyFileSearchPaths: [@"c:\test"]);
+            var provider2 = new DesktopStrongNameProvider(keyFileSearchPaths: [@"c:\test"]);
+
+            Assert.Equal(provider1, provider2);
+        }
+
+        [Fact]
+        public void InequalityUsingPath()
+        {
+            var tempDir = Temp.CreateDirectory();
+            var provider1 = new DesktopStrongNameProvider(tempPath: tempDir.Path);
+            var provider2 = new DesktopStrongNameProvider(tempPath: tempDir.Path + "2");
+
+            Assert.NotEqual(provider1, provider2);
+        }
+
+        [Fact]
+        public void InequalityUsingKeyFileSearchPaths()
+        {
+            var tempDir = Temp.CreateDirectory();
+            var provider1 = new DesktopStrongNameProvider(keyFileSearchPaths: [@"c:\test"]);
+            var provider2 = new DesktopStrongNameProvider(keyFileSearchPaths: [@"c:\test2"]);
+
+            Assert.NotEqual(provider1, provider2);
+        }
```

Comment:
Do you want to add one more test for case sensitivity also counting as unequal?

Reply:
Done!